staging/src/k8s.io/kubectl/pkg/util/resource/resource.go

```diff
@@ -54,6 +54,22 @@ func PodRequestsAndLimits(pod *corev1.Pod) (reqs, limits corev1.ResourceList) {
 			}
 		}
 	}
+
+	// If any container does not specify a limit that means the resource is not limited for the pod as a whole.  The
```

Comment:
Should we put this snippet to a new helper function then we could avoid duplicating code?

Reply:
I'm open to it.  There is quite a bit of duplicated code already and I didn't want to confuse things with this PR.

If the fix is acceptable, I can try to refactor into a reusable component, but I would need to figure out where that should go so both sections of the code can use it.

For now my main concern is making sure the issue's assertion is correct about this being a bug (I think it is too) and making sure others agree this is the correct fix for it.

Reply:
Understand